Content/ContentTypeResolver/SnippetSelectionResolver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+
+namespace Sulu\Bundle\HeadlessBundle\Content\ContentTypeResolver;
+
+
+use Sulu\Bundle\HeadlessBundle\Content\ContentView;
+use Sulu\Bundle\SnippetBundle\Snippet\SnippetResolver;
+use Sulu\Component\Content\Compat\PropertyInterface;
+
+class SnippetSelectionResolver implements ContentTypeResolverInterface
+{
+
+    /**
+     * @var SnippetResolver
+     */
+    private $snippetResolver;
+
+    /**
+     * SnippetSelectionResolver constructor.
+     * @param SnippetResolver $snippetResolver
+     */
+    public function __construct(SnippetResolver $snippetResolver)
+    {
+        $this->snippetResolver = $snippetResolver;
+    }
+
+    public static function getContentType(): string
+    {
+        return 'snippet_selection';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function resolve($data, PropertyInterface $property, string $locale, array $attributes = []): ContentView
+    {
+        if (null === $data) {
+            return new ContentView([]);
+        }
+
+        $content = $this->snippetResolver->resolve($data, $property, $locale, $attributes);
```

Comment:
I don't think that we can use the SnippetResolver here as we need to resolve every property itself e.g. the media need to be resolved over the MediaResolver and so on, but could not have a deeper look into just the first think which did come to my mind when scrolled over this PR.

Reply:
i think your right ! 
i change a lot to add the correct structure Resolver